services/console-proxy/server/src/com/cloud/consoleproxy/ConsoleProxySecureServerFactoryImpl.java

```diff
@@ -105,6 +122,87 @@ public void configure(HttpsParameters params) {
     }
 
     @Override
+    public Server createConsoleProxyServer(int port) throws IOException {
+
+        Server server = getServerWithContext(port);
+
+        /**
+         * add all context here
+         **/
+
+        // HTTP Configuration
+        HttpConfiguration http = new HttpConfiguration();
+        http.addCustomizer(new SecureRequestCustomizer());
+
+        // Configuration for HTTPS redirect
+        http.setSecurePort(443);
```

Comment:
Make sure you move any hard coded configuration to a static final config variable